### PR TITLE
Greenfield: Add image upload for app items

### PR DESCRIPTION
**Pull request by dennisreimann**
Upload endpoints for app item images. Follow-up to #6075. 

Tested to work with the app item editor.

### CODE REVIEW

**Comment by dstrukt** [comment]
YES!

I was going to make an issue for this, and you already have a PR 💥💥

**Comment by NicolasDorier** [comment]
@dennisreimann 

1. There isn't any UploadImage API for any other data stucture (for example store), as such, it shouldn't be expected for the app.
2. Instead, you should do like we do in Store: Use the `UnresolvedUri` type.

A user who wants to update will need to upload the file via the API to the file service, then reference it through an UnresolvedUri.

**Comment by dennisreimann** [comment]
Hmm, but there are dedicated upload actions for the [store logo](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/Controllers/GreenField/GreenfieldStoresController.cs#L109) as well as the [user profile picture](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/Controllers/GreenField/GreenfieldUsersController.cs#L234).

**Comment by NicolasDorier** [comment]
@dennisreimann this really sucks, we don't want to add a separate routes everytimes we want to upload a logo.

Let me think about it.

**Comment by dennisreimann** [comment]
We could do it via the GreenfieldFilesController, but we'd have to relax the permission requirement. When I implemented it, I matched the original UI-based file action policies, which make it available to admins only.

File: BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs [hunk 1]
```diff
@@ -249,6 +250,63 @@ public async Task<IActionResult> GetAppTopItems(string appId, [FromQuery] int of
...
+
+        [HttpPost("~/api/v1/apps/{appId}/image")]
+        [Authorize(Policy = Policies.CanModifyStoreSettings, AuthenticationSchemes = AuthenticationSchemes.Greenfield)]
+        public async Task<IActionResult> UploadAppItemImage(string appId, IFormFile? file)
+        {
+            var app = await _appService.GetApp(appId, null, includeArchived: true);
+            var userId = _userManager.GetUserId(User);
+            if (app == null || userId == null) return AppNotFound();
+            
+            if (file is null)
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs:262]: Can you use `_fileService.UploadImage` created by https://github.com/btcpayserver/btcpayserver/pull/6254?

Comment (dennisreimann) [reply BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs:262]: There are more places, in which we can leverage the new method. Shall I clean up all of them in one go or in a follow-up PR?

File: BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs [hunk 2]
```diff
@@ -249,6 +250,63 @@ public async Task<IActionResult> GetAppTopItems(string appId, [FromQuery] int of
...
+                return this.CreateValidationError(ModelState);
+            
+            try
+            {
+                var storedFile = await _fileService.AddFile(file!, userId);
+                var fileData = new FileData
+                {
+                    Id = storedFile.Id,
+                    UserId = storedFile.ApplicationUserId,
+                    Uri = new UnresolvedUri.FileIdUri(storedFile.Id).ToString(),
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs:286]: Do not return this. The API is already confusing enough :D

**Comment by woutersamaey** [comment]
Are you uploading the files base64 encoded or using HTTP form encoding? From an API POV, the base64 way may be more convenient?

**Comment by dennisreimann** [comment]
HTTP form, that's also how we implemented the other upload endpoints.

**Pull request merged**